extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -188,6 +202,7 @@ export class AmpAd3PImpl extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    this.lifecycleReporter_.sendPing('layoutCallback');
```

Comment:
Ah, I see.  I misunderstood the point you were making.  Ok, that's reasonable.  I'll update the names in the morning.